xarray/core/groupby.py

```diff
@@ -158,6 +159,15 @@ def ndim(self):
     def values(self):
         return range(self.size)
 
+    @property
+    def shape(self):
+        return (self.size,)
+
+    def __getitem__(self, key):
```

Comment:
`formatting._get_indexer_at_least_n_items` returns a tuple so I unpack it here. This could alternatively be an `if` condition in `formatting.first_n_items` and `formatting.last_n_items` but I felt this might be cleaner. I guess we could isntead call `np.array(self.values)[key]`